slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/PlainSQLTest.scala

```diff
@@ -116,12 +33,68 @@ class PlainSQLTest extends AsyncTest[JdbcTestDB] {
       DBIO.fold((for {
         (id, name) <- List((1, "szeiger"), (0, "admin"), (2, "guest"), (3, "foo"))
       } yield sqlu"insert into USERS values ($id, $name)"), 0)(_ + _) shouldYield(4),
-      sql"select id from USERS".as[Int] shouldYield Set(0,1,2,3), //TODO Support `to` in Plain SQL Actions
-      userForID(2).head shouldYield User(2,"guest"), //TODO Support `head` and `headOption` in Plain SQL Actions
+      sql"select id from USERS".as[Int] shouldYield Set(0,1,2,3),
+      userForID(2).head shouldYield User(2,"guest"),
       s1 shouldYield List(1),
       s2 shouldYield List(2),
-      userForIdAndName(2, "guest").head shouldYield User(2,"guest"), //TODO Support `head` and `headOption` in Plain SQL Actions
-      userForIdAndName(2, "foo").headOption shouldYield None //TODO Support `head` and `headOption` in Plain SQL Actions
+      userForIdAndName(2, "guest").head shouldYield User(2,"guest"),
+      userForIdAndName(2, "foo").headOption shouldYield None
+    )
+  }
+
+  def testComposableQueries = ifCap(tcap.plainSql) {
+    val createTable: DBIO[Int] = sqlu"create table USERS2(ID int not null primary key, NAME varchar(255))"
+
+    def insertUser(id: Int, name: String) = sql"INSERT INTO USERS2 VALUES ($id, $name)".asUpdate
+
+    val users = Set((1, "szeiger"), (0, "admin"), (2, "guest"), (3, "foo"), (4, "bar"))
+    val insertAll: DBIO[Int] = DBIO.fold(users.map{case(i, s) => insertUser(i, s)}.toSeq, 0)(_ + _)
+
+
+    def getAllIDs = sql"SELECT id FROM USERS2".as[Int]
+
+    def getUsers(name: Option[String] = None) = {
+      val q = sql"SELECT id, name FROM USERS2"
+      (name map {n => q ++ sql"WHERE NAME LIKE ($n)" } getOrElse q).as[(Int, String)]
```

Comment:
I would also expect Plain SQL composition to support splicing so that this query could be written as `sql"#$q WHERE NAME..."`. A more convenient (but slightly non-orthogonal) version would also allow `sql"$q WHERE NAME..."`. In this content of a string interpolator this seems more natural to me than composition with `++`. WDYT?